ROMFS/px4fmu_common/init.d/rcS

```diff
@@ -528,6 +548,16 @@ else
 				fi
 			fi
 		fi
+
+		if [ $OUTPUT_MODE == fmu ]
+		then
+			if fmu mode_$FMU_MODE $FMU_ARGS
+			then
+			else
+				echo "FMU start failed" >> $LOG_FILE
+				tune_control play -m ${TUNE_ERR}
+			fi
+		fi
 	fi
```

Comment:
@bkueng , I just pushed up a commit to this PR that places this logic block a few lines lower. I did this to reveal what looks like a logic error when compared with a few lines above, if [ $OUTPUT_MODE != fmu ].

Because if [ $OUTPUT_MODE != fmu ] is contained in the else statement, I think it could be taken out of the if statement that contains is, then we would have if [ $OUTPUT_MODE != fmu ] followed by if [ $OUTPUT_MODE == fmu ] that do exactly the same thing.

Let me know your thoughts on this!

Reply:
The above fmu start is also inside the else of `if [ $IO_PRESENT == yes ]`, which means there is a case where the fmu is not started (if `$IO_PRESENT == yes` && `$OUTPUT_MODE != fmu`) and thus we cannot simply drop/merge it.
Did I understand you correctly?
Also I think the startup ordering between io and fmu is important, we'd have to double-check that if we change it.